packages/expo-cli/src/commands/build/ios/appleApi/authenticate.js

```diff
@@ -52,22 +53,32 @@ async function _promptForAppleId({ appleId }) {
 
   log(wrap(chalk.bold('The password is only used to authenticate with Apple and never stored.')));
 
-  const appleIdQuestions = [
-    {
-      type: 'input',
-      name: 'appleId',
-      message: `Apple ID:`,
-      validate: nonEmptyInput,
-      when: () => !appleId,
-    },
+  if (!appleId) {
+    ({ appleId } = await prompt(
+      {
+        type: 'input',
+        name: 'appleId',
+        message: `Apple ID:`,
+        validate: nonEmptyInput,
+      },
+      {
+        nonInteractiveHelp: 'Pass your Apple ID using the --apple-id flag.',
+      }
+    ));
+  }
+  let { appleIdPassword } = await prompt(
     {
       type: 'password',
       name: 'appleIdPassword',
-      message: answer => `Password (for ${appleId || answer.appleId}):`,
+      message: answer => `Password (for ${appleId}):`,
       validate: nonEmptyInput,
     },
-  ];
-  return { appleId, ...(await prompt(appleIdQuestions)) };
+    {
+      nonInteractiveHelp:
+        'Pass your Apple ID password using the EXPO_APPLE_PASSWORD environment variable',
+    }
+  );
+  return { appleId, appleIdPassword };
```

Comment:
Changes in this function seem redundant to me. You could only add `nonInteractiveHelp` fields to the question objects.

Reply:
`nonInteractiveHelp` is not a property of the Inquirer.js questions, but a separate option handled by the `prompt` function. That's why I split the prompt to two separate calls.